Sources/StreamChat/StateLayer/UserSearch.swift

```diff
@@ -61,14 +61,9 @@ public class UserSearch {
     
     private func search(query: UserListQuery, pagination: Pagination) async throws -> [ChatUser] {
         let query = query.withPagination(pagination)
-        let task = Task {
-            let users = try await userListUpdater.fetch(userListQuery: query, pagination: pagination)
-            try Task.checkCancellation()
-            return users
-        }
-        await state.setActiveTask(task, query: query)
-        let users = try await task.value
-        await state.setUsers(users, for: query, pagination: pagination)
```

Comment:
Can you clarify in the PR description why this did not work? 🙏

Reply:
The problem with this one is if you write code like this in the UI:
```swift
state = .loading
do {
  try await userSearch.search(term: "name")
  state = .idle
} catch {
  state = .failure(error)
}
```
Then `Task.checkCancellation()` throws an error when the previous request has been cancelled. Then one needs to explicitly ignore the cancellation which feels like might cause too many support headache.

So for now, I just removed the cancellation part. Downside is that, we do not cancel requests what are irrelevant.

Since the search method also returns `[ChatUser]` then if felt incorrect to internally cancel and return an empty array.

Open for ideas. Some options:
1. We could add documentation about the cancellation and still cancel old requests
2. No cancellation, just discarding irrelevant query results

Reply:
As discussed on the tech meeting - I would go with 3) for now - request cancellation is a big topic, that ideally we would like to handle on the API client level.